_std/macros/switched_obj.dm

```diff
@@ -0,0 +1,11 @@
+//all switched objects in ID-based switched object groups should have:
+//- a call to ADD_SWITCHED_OBJECT in initialize, using a switched object category define, and a call to REMOVE_SWITCHED_OBJECT in disposing.
+//- an "id" var (typically null by default, and varedit-able in maps; used to group the switchable objects)
+//- a "toggle" proc that accepts a true/false state (handled in decorations.dm currently)
+//if you miss any of these, things are probably gonna break
+
+#define ADD_SWITCHED_OBJ(cat) if(!switched_objs[cat]) { switched_objs[cat] = list() }; \
+	if(!switched_objs[cat][src:id]) { switched_objs[cat][src:id] = list() }; \
+	switched_objs[cat][src:id] += src
```

Comment:
Why is this using `:`? `src` should always have a known correct type anyway

Reply:
I think my previous implementation that wasn't using macros freaked out if it was using . instead of : but I have no objection to switching if it works here.